tutorials/seqcopy/README.md

```diff
@@ -0,0 +1,140 @@
+(`dev.source`, `dev.target`). The sequences that were generate look like this:
+
+```
+2 3 5 5 4 6 7 0 3 8 10 9 3 6
+9 9 1 5 3 0 5 4 0 8 8 5 7 7 8 7 3 1 0
+9 1 9 7 9 1 9 9 9 3 9 3 2 8 0 1 6 10 4 3 1 9 2 7 1 5 7 7 5 5
+2 1 4 10 7 7 7 2 10 9 4 9 9 7 8 4 10 6 8 2 6 7 5 3 2
+4 6 0 7 8 8 6 3 4 10 2 10 6 9 5 3
+8 0 5 4 1 8 0 8 7 4 4 0 0 9 5 8 9
+```
+
+## 2. Training
+
+Now that we have some training data to play with we can train our model. Start training by running
+the following command:
+
+```bash
+python3 -m sockeye.train -s train.source \
+                         -t train.target \
+                         -vs dev.source \
+                         -vt dev.target \
+                         --num-embed 32 \
+                         --rnn-num-hidden 64 \
+                         --attention-type dot \
+                         --use-cpu \
+                         --metrics perplexity accuracy \
+                         --max-num-checkpoint-not-improved 3 \
+                         -o seqcopy_model
+```
+
+This will train a 1-layer RNN model with a bidirectional LSTM as the encoder side and a uni-directional LSTM
+as the decoder.
+The RNNs have 64 hidden units and we learn embeddings of size 32.
+Looking at the log we can see that our training data was assigned to buckets according to their
+lengths. Additionally, Sockeye will take care of correctly padding sequences and masking relevant parts of the network,
+in order to deal with sequences of variable length.
+
+
+### Metrics and checkpointing
+During training Sockeye will print relevant metrics on both the training and the validation data.
+The metrics can be chosen using the `--metrics` parameter.
+Validation metrics are evaluated every time we create a checkpoint.
+During checkpointing the current model parameters are saved into the model directory and current validation scores
+are evaluated.
+By default Sockeye will create a checkpoint every 1000 updates.
+This can be adjusted through the `--checkpoint-frequency` parameter.
+
+From the log you can see that initially the accuracy is around 0.1:
+```bash
+...
+[INFO:sockeye.training] Training started.
+[INFO:sockeye.callback] Early stopping by optimizing 'perplexity'
+[INFO:root] Epoch[0] Batch [50]  Speed: 683.23 samples/sec perplexity=14.104128 accuracy=0.092011
+[INFO:root] Epoch[0] Batch [100] Speed: 849.97 samples/sec perplexity=13.036482 accuracy=0.096760
+...
+```
+With a vocabulary of size 10 this essentially means that the model is guessing randomly.
+As training progresses we see that after around 14 epochs the accuracy goes up to ~1.0 and the perplexity down to ~1.0.
+Sockeye performs early stopping based on the validation metrics tracked when checkpointing.
+Once the validation metrics have not improve for several checkpoints the training is stopped.
+How many validation non-improved checkpoints to tolerate can be set adjusted `--max-num-checkpoint-not-improved`.
```

Comment:
The number of tolerated non-improving checkpoints can be adjusted ...

Reply:
changed